plugins/search-backend-node/src/engines/LunrSearchEngine.test.ts

```diff
@@ -75,6 +75,75 @@ const getActualIndexer = (engine: SearchEngine, index: string) => {
   return engine.getIndexer(index);
 };
 
+describe('stopword-testing', () => {
+  let testLunrSearchEngine: SearchEngine;
+
+  beforeEach(() => {
+    testLunrSearchEngine = new LunrSearchEngine({ logger: getVoidLogger() });
+    jest.clearAllMocks();
+  });
+
+  it('test with stop words', async () => {
+    // Add another test case for builds-at-seek, really wondering why cannot reproduce
+    // const mockDocuments = [
+    //   {
+    //     title: 'discover-and-career-services',
+    //     text: 'discover-and-career-services',
+    //     location: 'test/location',
+    //   },
+    // ];
+    const mockDocuments = [
+      {
+        title: 'builds-at-seek',
+        text: 'builds-at-seek',
+        location: 'test/location',
+      },
+    ];
+
+    const indexer = await getActualIndexer(testLunrSearchEngine, 'test-index');
+
+    await TestPipeline.withSubject(indexer)
+      .withDocuments(mockDocuments)
+      .execute();
+
+    // const mockedSearchResult = await testLunrSearchEngine.query({
+    //   term: 'discover',
+    //   filters: {},
+    // });
+    const mockedSearchResult = await testLunrSearchEngine.query({
```

Comment:
I don't think we're exercising the right code path here, which is why you're not seeing the expected failure. The query we use has filters. It's the logic that takes these filters and turns them in to query clauses that has the bug. I think something like:

```
    const mockDocuments = [
      {
        title: 'builds-at-seek',
        text: 'builds-at-seek',
        location: 'test/builds-at-seek',
      },
    ];
```

And then 

```
const mockedSearchResult = await testLunrSearchEngine.query({
      term: 'builds-at-seek',
      filters: {
        "location": "builds-at-seek"
      },
    });
```

might have a better chance of reproducing the issue. Note that I just made up that filter syntax, it might be wrong. A better way of identifying the correct syntax there would be to debug this issue locally, and see what filters Backstage uses when constructing the query that is failing.